extensions/windows-event-log/wel/LookupCacher.h

```diff
@@ -0,0 +1,48 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <chrono>
+#include <functional>
+#include <mutex>
+#include <string>
+#include <unordered_map>
+#include <utility>
+
+namespace org::apache::nifi::minifi::wel {
+
+class LookupCacher {
```

Comment:
It is not completely general as it only handles `string -> string` functions.  Let's keep this open for a while; if nobody objects to moving it to libminifi, I will move it.

Reply:
I think it could be a generic utility, but for that it would need to be able to handle a configurable max size and evict oldest entries when we exceed this limit (an LRU cache)

Reply:
Sounds good, as I previously mentioned I do not insist on this, it can be handled in a separate jira if it is viable.

Reply:
I would still keep it in wel as long as it's only used here. We have too many utilities that are general in theory, but are only ever used in one extension.